### PR TITLE
US-465 Prevent undefined index error

### PR DESCRIPTION
**Pull request by alexandre-xavier-auto1**
- [x] Task: [US-465](https://wkdauto.atlassian.net/browse/US-465)

### Change

Prevent undefined index on [EndpointRouteDescriber.php#L104](https://github.com/auto1-oss/service-api-handler-bundle/blob/master/ApiDoc/EndpointRouteDescriber.php#L104) by adding `array_key_exists` check

```php
$request  = $this->controllerToRequestMapping[$controller];
```

### CODE REVIEW

File: ApiDoc/EndpointRouteDescriber.php [hunk 1]
```diff
@@ -97,6 +97,10 @@ private function getEndpoint(Route $route)
     {
         $controller = $route->getDefault('_controller');
 
+        if (!\array_key_exists($controller, $this->controllerToRequestMapping)) {
+            return null;
+        }
+
         if (in_array($controller, $this->controllerToRequestMapping)) {
```
Comment (daydiff) [inline ApiDoc/EndpointRouteDescriber.php]: I believe this block should be removed. The previous condition is a correct replace for that.

**Review by daydiff — CHANGES_REQUESTED** [review]

File: ApiDoc/EndpointRouteDescriber.php (hunk 1 above)
Comment (alexandre-xavier-auto1) [reply ApiDoc/EndpointRouteDescriber.php]: I also though about removing it, what do you think @sklabinsky and @Dropaq?

Comment (Dropaq) [reply ApiDoc/EndpointRouteDescriber.php]: The statement is simply wrong. First searches by key, second by value. Figure which one you need

Comment (alexandre-xavier-auto1) [reply ApiDoc/EndpointRouteDescriber.php]: The second check was removed considering that controller is the key part of the mapping.

So, `$this->controllerToRequestMapping`  receives [`auto1.api_handler.controller_request_mapping`](https://github.com/auto1-oss/service-api-handler-bundle/blob/master/DependencyInjection/CompilerPass/EndpointRouterCompilerPass.php#L83)  parameter which [stores controller part as keys](https://github.com/auto1-oss/service-api-handler-bundle/blob/master/DependencyInjection/CompilerPass/EndpointRouterCompilerPass.php#L100), therefore there is no need of checking values.

**Review by LoopKarma — APPROVED** [review]

**Review by sklabinsky — APPROVED** [review]

**Pull request merged**